### DISCUSSION TITLE
[Android] Empty screen when creating a new MainPage instead of navigating to it in App.xaml.cs.

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#17277] [Q&A] [answered]
It works on Windows and Desktop, but fails on Android.

Original code:
```csharp
if (RootFrame!.Content == null)
{
    RootFrame.Navigate(typeof(MainPage), args.Arguments);
}
```
Modified:
```csharp
if (RootFrame!.Content == null)
{
    RootFrame.Content = new MainPage();
}
```
![screen1](https://github.com/unoplatform/uno/assets/24267381/d5b19567-69ee-4445-8720-96619918fb8f)  ![screen2](https://github.com/unoplatform/uno/assets/24267381/f500f68e-90e6-44b7-b3b7-da44995359b8)

**Comment by jeromelaban** [comment, marked as answer]
The `Frame` control does not support changing its content this way on android, specifically. If you want to change the root, set the MainPage directly to the `Window.Content` instead.
